src/miral/application_selector.cpp

```diff
@@ -271,7 +275,13 @@ auto ApplicationSelector::advance(bool reverse, bool within_app) -> Window
 
     auto next_state_to_preserve =  tools.info_for(next_window.value()).state();
     tools.select_active_window(next_window.value());
-    restore_state = next_state_to_preserve;
+
+    // Don't re-minimize windows selected with alt tab on focus loss
+    if(next_state_to_preserve != mir_window_state_minimized)
+        restore_state = next_state_to_preserve;
+    else
+        restore_state.reset();
```

Comment:
I think you're looking at it from the point of view of tabbing between open apps. I was looking at it from the point of view of _selecting_ between open apps. So I expected minimized windows to not automatically minimize again. Thinking about it again, I think I was wrong

Reply:
Yeah I think that this should not be the case, since they should go back to being hidden.

That being said, `ApplicationSelector` stinks and should be replaced :smile: